### PR TITLE
[NCL-1685] storing results of a failed build

### PR DESCRIPTION
**Pull request by michalszynkiewicz**

### CODE REVIEW

File: build-coordinator/src/main/java/org/jboss/pnc/core/builder/datastore/DatastoreAdapter.java [hunk 1]
```diff
@@ -106,11 +106,41 @@ public void storeResult(BuildTask buildTask, BuildResult buildResult) throws Dat
             log.debugf("Storing results of buildTask [%s] to datastore.", buildTask.getId());
             datastore.storeCompletedBuild(buildRecordBuilder, buildTask.getBuildRecordSetIds());
         } catch (Exception e) {
-            storeResult(buildTask, e);
+            storeErroneousResult(buildTask, e);
+        }
+    }
+
+
+    public void storeFailedResult(BuildTask buildTask, BuildResult buildResult) throws DatastoreException {
```
Comment (matejonnet) [inline build-coordinator/src/main/java/org/jboss/pnc/core/builder/datastore/DatastoreAdapter.java]: Does it make sense to have another method for storing the failed build? 
Isn't the result in both cases mostly the same?

Comment (michalszynkiewicz) [inline build-coordinator/src/main/java/org/jboss/pnc/core/builder/datastore/DatastoreAdapter.java]: There's indeed some code duplucation. I'll reduce it. For now I think that two methods are clearer, but I'll try out merging too. (after some tests start to show that it works ;) )

Thank you for reviewing the changes!

Comment (matejonnet) [inline build-coordinator/src/main/java/org/jboss/pnc/core/builder/datastore/DatastoreAdapter.java]: Once you merge in public method others can use it and "for now" statement turns in long removal procedure.
I also wonder if changes on coordinator side are required for this bug fix?

Comment (michalszynkiewicz) [inline build-coordinator/src/main/java/org/jboss/pnc/core/builder/datastore/DatastoreAdapter.java]: Do you mean BuildCoordinator class itself or do you count DatastoreAdapter as a coordinator side too?
If it's the latter - yes they are required.

Comment (michalszynkiewicz) [inline build-coordinator/src/main/java/org/jboss/pnc/core/builder/datastore/DatastoreAdapter.java]: Here a version with one method for storing failed and successful builds:

**Comment by michalszynkiewicz** [comment]
replaced with: https://github.com/project-ncl/pnc/pull/890

**Pull request closed without merging**